ui/components/app/wallet-overview/token-overview.js

```diff
@@ -80,6 +90,30 @@ const TokenOverview = ({ className, token }) => {
       }
       buttons={
         <>
+          {isBuyable && (
+            <IconButton
+              className="token-overview__button"
+              Icon={BuyIcon}
+              label={t('buy')}
+              onClick={() => {
+                trackEvent({
+                  event: 'Clicked Deposit: Token',
+                  category: EVENT.CATEGORIES.NAVIGATION,
+                  properties: {
+                    action: 'Home',
+                    legacy_event: true,
+                  },
+                });
+                dispatch(
```

Comment:
With the modern modal approach, this can be replaced by updating the value of a local state variable which will trigger the modal in the rendering block.

Reply:
reply here: https://github.com/MetaMask/metamask-extension/pull/15551#discussion_r944673984